adapter/memQuota/rollingWindow.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memQuota
+
+import (
+	"sync"
+)
+
+// A single allocation, carrying its amount and expiration time
```

Comment:
Yes, I looked at a few rate limiters. They seem to all suffer from the '11th hour' problem. That is, if you have a QPS limit of 10, it's possible for someone to allocate 10 units at 0.9s and another 10 units at 1.1s.

My approach is a true rolling window, allowing no more than 10 units in any 1 second interval.

On top of that, my version is considerably more efficient in memory & time :-)